utils/writeable/src/lib.rs

```diff
@@ -121,15 +147,17 @@ pub trait Writeable {
         let mut output = String::with_capacity(self.write_len().capacity());
         self.write_to(&mut output)
             .expect("impl Write for String is infallible");
+        if !self.write_len().is_exact() {
+            output.shrink_to_fit();
```

Comment:
@Manishearth Do you think it's good for us to `realloc` here in this lower-level function?

Reply:
I would not.

Reply:
Why?

Reply:
I've removed it because I want to get this merged, but I'm still interested in your reasoning.